### PR TITLE
add support for pod disruption budget

### PR DESCRIPTION
**Pull request by jaypatrickhoward**
#### What this PR does / why we need it:
Adds support for [Pod Disruption Budget](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.11/#poddisruptionbudget-v1beta1-policy).

#### Checklist
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

File: stable/metrics-server/Chart.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ apiVersion: v1
 appVersion: 0.3.2
 description: Metrics Server is a cluster-wide aggregator of resource usage data.
 name: metrics-server
-version: 2.6.0
+version: 2.6.1
```
Comment (olemarkus) [inline stable/metrics-server/Chart.yaml]: Could you bump minor version here?

File: stable/metrics-server/values.yaml [hunk 2]
```diff
@@ -90,3 +90,7 @@ service:
   annotations: {}
   port: 443
   type: ClusterIP
+
+# The values to set in the PodDisruptionBudget spec (minAvailable/maxUnavailable)
+# If not set then a PodDisruptionBudget will not be created
+podDisruptionBudget:
```
Comment (olemarkus) [inline stable/metrics-server/values.yaml:94]: Could you add a (commented out) example of what is users should add here? Right now, one has to look at the template.

**Review by olemarkus — COMMENTED** [review]
Thanks for this.

I wonder if not the way that e.g prometheus-operator implements this is better though. Where you have to specify something like this in the values file:
``` podDisruptionBudget:
    enabled: false
    minAvailable: 1
    maxUnavailable: ""
```

Less flexible, but more explicit and user friendly.

File: stable/metrics-server/values.yaml (hunk 2 above)
Comment (jaypatrickhoward) [reply stable/metrics-server/values.yaml:94]: Will do.  I have another nearly-identical PR pending for the kubernetes-dashboard chart.  Do you advise preemptively making the same changes there?

Comment (jaypatrickhoward) [reply stable/metrics-server/values.yaml:94]: To clarify:  all the changes, not just the additional documentation in values.yaml.

Comment (olemarkus) [reply stable/metrics-server/values.yaml:94]: I would advice that yes. The kubernetes-dashboard maintainer may have other preferences though.

Comment (jaypatrickhoward) [reply stable/metrics-server/values.yaml:94]: I went ahead and made the changes over there as well.  No response yet, but I'm happy to do it whichever way they prefer.

**Pull request merged**